### PR TITLE
fix Issue 15565 - Forward reference error with namespaces

### PR DESCRIPTION
**Pull request by WalterBright**
https://issues.dlang.org/show_bug.cgi?id=15565

### CODE REVIEW

**Comment by WalterBright** [comment]
This is critical for Manu's work.

File: src/nspace.d [hunk 1]
```diff
@@ -63,26 +51,64 @@ public:
...
+            sc.pop();
+        }
+    }
+
+    override void semantic(Scope* sc)
+    {
+        if (semanticRun >= PASSsemantic)
+            return;
+        semanticRun = PASSsemantic;
+        static if (LOG)
```
Comment (andralex) [inline src/nspace.d]: we shouldn't have all uppercase names

**Comment by andralex** [comment]
Auto-merge toggled on

File: src/nspace.d (hunk 1 above)
Comment (jacob-carlborg) [inline src/nspace.d]: I'm pretty sure this is an artifact of translating a preprocessor symbol to D.

**Pull request merged**